mscore/qmltheme.h

```diff
@@ -0,0 +1,71 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#ifndef MSF_QMLTHEME_H
+#define MSF_QMLTHEME_H
+
+#include <QObject>
+#include <QPalette>
+#include <QColor>
+
+namespace msf {
+
+class QmlTheme : public QObject
+{
+    Q_OBJECT
+
+#define COLOR_PROPERTY(name, role) \
+    Q_PROPERTY(QColor name READ get##name NOTIFY themeChanged) \
+    QColor get##name() const { return _palette.color(role); }
+
+    COLOR_PROPERTY(window, QPalette::Window)
```

Comment:
I have initially added all this `COLOR_PROPERTY` stuff because of the need to support Qt 5.9. Since Qt 5.10 it is possible various [`palette`](https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#palette-prop) properties and [`palette`](https://doc.qt.io/qt-5/qml-palette.html#qtquickcontrols2-palette) type in general can be used instead. Should perhaps the new framework make use of this (in case no Qt 5.9 support is planned)?

Reply:
Actually, we decided that we would not use QT palettes, they have a number of shortcomings